### PR TITLE
Only clone the metric in the one place relabelling needs it.

### PR DESCRIPTION
**Pull request by brian-brazil**
This cuts ~17% off memory allocations related to ingesting data
in a basic setup.

### CODE REVIEW

File: relabel/relabel.go [hunk 1]
```diff
@@ -26,17 +26,15 @@ import (
 // Process returns a relabeled copy of the given label set. The relabel configurations
 // are applied in order of input.
 // If a label set is dropped, nil is returned.
+// May return the input labelSet modified .
```
Comment (beorn7) [inline relabel/relabel.go]: Spurious blank.

**Comment by beorn7** [comment]
Looks great in principle. I'd just feel much more comfortable if somebody familiar with the original code (and why we did it that way) could look at it. @fabxc ?

File: relabel/relabel.go (hunk 1 above)
Comment (brian-brazil) [inline relabel/relabel.go]: Fixed

**Comment by brian-brazil** [comment]
I think I was the one to do it this way, and I left it like this rather than optimise to keep it simple.

File: retrieval/target.go [hunk 2]
```diff
@@ -304,7 +304,7 @@ func populateLabels(lset model.LabelSet, cfg *config.ScrapeConfig) (res, orig mo
 		}
 	}
 
-	preRelabelLabels := lset
+	preRelabelLabels := lset.Clone()
```
Comment (grobie) [inline retrieval/target.go]: What about only cloning if `len(cfg.RelabelConfigs) > 0`?

Comment (grobie) [inline retrieval/target.go]: I see that we modify the label set afterwards to remove the meta labels. I guess for UI purposes we'll always have to carry around the two sets (with and without meta labels).

**Comment by grobie** [comment]
LGTM

**Pull request merged**